Sources_v3/StreamChatUI/ChatChannel/ChatMessageImageGallery.swift

```diff
@@ -0,0 +1,220 @@
+                
+                previews[1].trailingAnchor.constraint(equalTo: trailingAnchor),
+                previews[1].topAnchor.constraint(equalTo: topAnchor),
+                previews[1].bottomAnchor.constraint(equalTo: bottomAnchor),
+                previews[1].widthAnchor.constraint(equalTo: previews[0].widthAnchor)
+            ],
+            [
+                previews[0].leadingAnchor.constraint(equalTo: leadingAnchor),
+                previews[0].topAnchor.constraint(equalTo: topAnchor),
+                previews[0].bottomAnchor.constraint(equalTo: bottomAnchor),
+                previews[0].widthAnchor.constraint(equalTo: widthAnchor, multiplier: 0.5, constant: anchorSpacing),
+                
+                previews[1].topAnchor.constraint(equalTo: topAnchor),
+                previews[1].trailingAnchor.constraint(equalTo: trailingAnchor),
+                previews[1].heightAnchor.constraint(equalTo: heightAnchor, multiplier: 0.5, constant: anchorSpacing),
+                previews[1].widthAnchor.constraint(equalTo: previews[0].widthAnchor),
+                
+                previews[2].trailingAnchor.constraint(equalTo: previews[1].trailingAnchor),
+                previews[2].heightAnchor.constraint(equalTo: previews[1].heightAnchor),
+                previews[2].widthAnchor.constraint(equalTo: previews[1].widthAnchor),
+                previews[2].bottomAnchor.constraint(equalTo: bottomAnchor)
+            ],
+            [
+                previews[0].leadingAnchor.constraint(equalTo: leadingAnchor),
+                previews[0].topAnchor.constraint(equalTo: topAnchor),
+                previews[0].widthAnchor.constraint(equalTo: widthAnchor, multiplier: 0.5, constant: anchorSpacing),
+                previews[0].heightAnchor.constraint(equalTo: widthAnchor, multiplier: 0.5, constant: anchorSpacing),
+                
+                previews[1].topAnchor.constraint(equalTo: topAnchor),
+                previews[1].trailingAnchor.constraint(equalTo: trailingAnchor),
+                previews[1].heightAnchor.constraint(equalTo: previews[0].heightAnchor),
+                previews[1].widthAnchor.constraint(equalTo: previews[0].widthAnchor),
+                
+                previews[2].leadingAnchor.constraint(equalTo: leadingAnchor),
+                previews[2].heightAnchor.constraint(equalTo: previews[0].heightAnchor),
+                previews[2].widthAnchor.constraint(equalTo: previews[0].widthAnchor),
+                previews[2].bottomAnchor.constraint(equalTo: bottomAnchor),
+                
+                previews[3].trailingAnchor.constraint(equalTo: trailingAnchor),
+                previews[3].heightAnchor.constraint(equalTo: previews[0].heightAnchor),
+                previews[3].widthAnchor.constraint(equalTo: previews[0].widthAnchor),
+                previews[3].bottomAnchor.constraint(equalTo: bottomAnchor)
+            ]
+        ]
```

Comment:
Can't this be done using 3 stack views and simply by making the previews without content hidden? Something like:
```
┌───────────────────────────────────────────────────────────┐
│ ┌──────────────────────────┐ ┌──────────────────────────┐ │
│ │┌────────────────────────┐│ │┌────────────────────────┐│ │
│ ││                        ││ ││                        ││ │
│ ││                        ││ ││                        ││ │
│ ││                        ││ ││                        ││ │
│ ││        preview 1       ││ ││       preview 3        ││ │
│ ││                        ││ ││                        ││ │
│ ││                        ││ ││                        ││ │
│ │└────────────────────────┘│ │└────────────────────────┘│ │
│ │┌────────────────────────┐│ │┌────────────────────────┐│ │
│ ││                        ││ ││                        ││ │
│ ││                        ││ ││                        ││ │
│ ││         preview 2      ││ ││       preview 4        ││ │
│ ││                        ││ ││                        ││ │
│ ││                        ││ ││                        ││ │
│ │└────────────────────────┘│ │└────────────────────────┘│ │
│ └──────────────────────────┘ └──────────────────────────┘ │
└───────────────────────────────────────────────────────────┘
```

Reply:
Used constraints for the same reason as in message bubble: "stack-view based cells the message list behaves much more glitchy while scrolling"